lib/chef/knife/bootstrap.rb

```diff
@@ -549,7 +549,28 @@ def run
 
         $stdout.sync = true
         register_client
-        connect!
+        begin
+          connect!
+        rescue Train::Transports::SSHFailed => e
+          if e.message =~ /fingerprint (\S+) is unknown for "(.+)"/
+            fingerprint = $1
+            hostname,ip = $2.split(',')
+            puts "The authenticity of host '#{hostname} (#{ip})' can't be established."
+            # TODO: convert the SHA256 base64 value to hex with colons
+            # 'ssh' example output:
+            # RSA key fingerprint is e5:cb:c0:e2:21:3b:12:52:f8:ce:cb:00:24:e2:0c:92.
+            # ECDSA key fingerprint is 5d:67:61:08:a9:d7:01:fd:5e:ae:7e:09:40:ef:c0:3c.
+            puts "fingerprint is #{fingerprint}."
+            ui.confirm("Are you sure you want to continue connecting")
```

Comment:
Does this raise if they pick no? If not, we need to check the return value before proceeding.

Reply:
`#confirm` will exit 3 if they say no.